Survey/Sources/Presentation/Modules/Home/HomeView.swift

```diff
@@ -166,11 +181,22 @@ struct HomeViewPreView: PreviewProvider {
 
     static var previews: some View {
         let viewModel = HomeViewModel(
-            useCase: HomeUseCase(
+            homeUseCase: HomeUseCase(
                 surveyRepository: SurveyRepository(
                     api: AuthenticationNetworkAPI()
                 )
-            )
+            ),
+            userUseCase: UserUseCase(
+                userRepository: UserRepository(
+                    api: AuthenticationNetworkAPI()
+                )
+            ),
+            logoutUserUseCase: UserUseCase(
```

Comment:
I think creating two identical protocols in the same file is easy to misused. Other dev might end up using userUseCase for logout and logoutUseCase for getUser because the protocol allow them to.
I suggest you create another signature func for `NetworkAPIProtocol` 
```
    func performEmptyRequest<T: Decodable & Encodable>(_ configuration: RequestConfiguration) -> Future<T, Error> {}
```
and call the special variant of `request` in this block.

This way you don't have to add extra use case every time you need both the normal and empty response.

There are other place you can do the same (such as having two `NetworkAPIProtocol` in `Repository` but I think modifying `NetworkAPIProtocol` directly saves you the most code change in the long run.

Reply:
> Other dev might end up using userUseCase for logout and logoutUseCase for getUser because the protocol allow them to. I suggest you create another signature func for NetworkAPIProtocol

That's good approach. It helps reducing much change for this feature and good for the long run. Thank you so much for your advice 🙏 .